server/repository/src/db_diesel/changelog/changelog.rs

```diff
@@ -522,6 +523,30 @@ fn create_filtered_outgoing_sync_query(
         .select(store::id.nullable())
         .into_boxed();
 
+    // Ideally this would be by changelog name_link_id, but that has an FK constraint
+    // requiring all names to exist on OMS central, which currently isn't the case.
+    // Instead, for visible patient sync - filter changelogs by record id of vaccinations
+    // for visible patients
+    // Bit of a hack, subquery unlikely to scale well - bring on v7 sync :cry:
```

Comment:
After discussion with @andreievg - think this is better than migrating entire changelog table to remove the reference constraint, could be a lot of records.

We should keep an eye on sync performance where vaccinations are being used...